agent/websockets/shim.go

```diff
@@ -122,6 +122,23 @@ const (
         }
       }
 
+      self.pendingMessages = [];
+      self.pushing = false;
+      self.push = function() {
+         if (self.pushing) {
```

Comment:
Do we need to worry about thread safety and adding a mutex here? (looking at [StackOverflow](https://stackoverflow.com/a/124832/21539), the answer seems to be "no"...)

Reply:
My searching also lead me to believe the answer is "no".

Also, there do not seem to be any cross-browser mutex options for Javascript that I could find, so I think we just have to trust that the answer really is "no" (which does seem to be the case).